sdk/metric/meter_test.go

```diff
@@ -2427,3 +2427,16 @@ func TestDuplicateInstrumentCreation(t *testing.T) {
 		})
 	}
 }
+
+func TestMeterProviderDelegation(t *testing.T) {
```

Comment:
how does this fail before your fix?  I don't see any checks

Reply:
it panics due to null pointer

Reply:
If we had the return suggested by @blkt then I think it would correctly return an error instead of panicking

Reply:
yea it won't panic but I will need to register a global error handler in that case

Reply:
I have added the return to remove the panic, added some additional error checks and added a global error handler to the unit test. Let me know if I am missing anything else.

Reply:
great, thanks!